sonic-net/sonic-buildimage
# image_config: copp: Enable rate limiting for bgp, lacp, dhcp, lldp, macsec and udld

#### Why I did it
It was observed that a flood of DHCP packets without rate-limiting can cause BGP flaps or lacp keepalive losses.
This change attempts to prevent or reduce such BGP flaps by enabling appropriate rate-limiting in SONiC for all traffic types.
##### Work item tracking
- Microsoft ADO **17964421**:

#### How I did it
Set a reasonable CIR/CBS value of 300 for queue4_group3 (dhcp, lldp, macsec) and 6000 for queue4_group1.
The value 300 was arrived at after testing with dhcp flooding using ptf (using multiple threads). Throttling at this rate was necessary to ensure that dhcp flooding does not cause BGP flaps.

#### How to verify it
1. Verified with this script running from ptf, that BGP flaps don't happen when CBS/CIR is set at 300 for queue4_group3.

        import threading
        from scapy.all import *
        
        def send_dhcp_discover(intf):
            dhcp_discover = Ether(dst='ff:ff:ff:ff:ff:ff',src=RandMAC()) \
                                /IP(src='1.1.1.1',dst='255.255.255.255') \
                                /UDP(sport=68,dport=67) \
                                /DHCP(options=[('message-type','discover'),('end')])
            sendp(dhcp_discover,count=100000,iface=intf)
        
        
        if __name__ == "__main__":
            t1 = threading.Thread(target=send_dhcp_discover, args=("eth1",))
            t2 = threading.Thread(target=send_dhcp_discover, args=("eth2",))
            t1.start()
            t2.start()
            t1.join()
            t2.join()

2. Verified on Arista-7260CX3-D108C8 running 202012 that the copp rule for queue4_group1 and queue4_group3 do NOT affect BGP packets. To verify this using PTF, the copp rules were modified to set the "CBS" and "CIR" for queue4_group1 and queue4_group3 at 600pps and 50k packets each of "BGP open" and "DHCP Discover" were simultaneously sent from the same PTF port to the DUT. It was verified using "show c cpu" that packets are hitting the cpu queue at 1200 pps (double the configured CIR/CBS  for these packet types).  This helped conclude that throttling rate is per trap (or packet type) and not per queue.

3. Verified with updated sonic-mgmt tests (https://github.com/sonic-net/sonic-mgmt/pull/8199) on broadcom and mellanox platforms that these traffic types are rate-limited.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)



- [x]  20220531.33
- [x] 20201231.107

#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
